subprojects/instant-execution/src/main/kotlin/org/gradle/instantexecution/serialization/Logging.kt

```diff
@@ -112,3 +112,19 @@ fun IsolateContext.logPropertyProblem(action: String, problem: PropertyProblem)
     logger.debug("configuration-cache > failed to {} {} because {}", action, problem.trace, problem.message)
     onProblem(problem)
 }
+
+
+internal
+inline fun <T : WriteContext, R> T.withDebugFrame(name: () -> String, writeAction: T.() -> R): R {
+    val frameName: String? = if (logger.isDebugEnabled) name() else null
+    try {
+        frameName?.let {
+            logger.debug("[{}] {\"type\":\"O\",\"frame\":\"{}\",\"at\":{}}", category, it, writePosition)
```

Comment:
maybe wrap the calls to `logger.debug` in a check that debug logging is enabled, as these calls generate a lot of garbage (via the varargs parameter array) that are unnecessary when it is not enabled.

Reply:
Thanks for the review, @adammurdoch.

The calls are guarded by `frameName != null` which is derived from `logger.isDebugEnabled`. I've benchmarked it against master and there's no degradation in non-debug mode.